apps/web/src/pages/templates/workflow/workflow/node-types/WorkflowNode.tsx

```diff
@@ -261,36 +199,63 @@ export function WorkflowNode({
             </When>
           </ActionWrapper>
         </Group>
+
         {!hasActiveIntegration && (
-          <Popover
+          <NodeErrorPopover
             opened={popoverOpened}
             withinPortal
             transition="rotate-left"
             transitionDuration={250}
             offset={theme.spacing.xs}
             target={<ErrorCircle data-test-id="error-circle" dark={theme.colorScheme === 'dark'} />}
-            title="Connect provider"
-            titleGradient="red"
-            description={`Please configure a ${CHANNEL_TYPE_TO_STRING[
-              channelKey
-            ]?.toLowerCase()} provider to send notifications over this channel`}
-            content={
-              <ConfigureProviderButton
+            titleIcon={<ProviderMissing />}
+            title={`${CHANNEL_TYPE_TO_STRING[channelKey]} provider is not connected`}
+            content={`Please configure or activate a provider instance for the ${CHANNEL_TYPE_TO_STRING[channelKey]} channel to send notifications over this node`}
+            actionItem={
+              <Button
                 onClick={() => {
                   segment.track(TemplateEditorAnalyticsEnum.CONFIGURE_PROVIDER_POPOVER_CLICK);
                   setIntegrationsModalVisible(true);
                   setPopoverOpened(false);
                 }}
               >
-                Configure
-              </ConfigureProviderButton>
+                Open integration store
+              </Button>
             }
           />
         )}
-        {hasActiveIntegration && stepErrorContent && (
-          <MantinePopover
+        {hasActiveIntegration && !primaryIntegration && isChannelStep && (
```

Comment:
`isChannelStep` flag would be true also for the chat integration, but chat, push, in-app channels don't have a primary provider functionality... so basically you have to check:
```
hasActiveIntegration && !primaryIntegration && (isEmailChannel || isSmsChannel)
``

Reply:
it works because the `primaryIntegration` returns the value for chat, push, in-app